internal/proxy/proxy_test.go

```diff
@@ -179,3 +179,30 @@ func TestClientInitialization(t *testing.T) {
 		})
 	}
 }
+
+func TestConfigDialerOpts(t *testing.T) {
+	tcs := []struct {
+		desc    string
+		config  proxy.Config
+		wantLen int
+	}{
+		{
+			desc:    "when there are no options",
+			config:  proxy.Config{},
+			wantLen: 0,
+		},
+		{
+			desc:    "when a token is present",
+			config:  proxy.Config{Token: "my-token"},
+			wantLen: 1,
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.desc, func(t *testing.T) {
+			if got := tc.config.DialerOpts(); tc.wantLen != len(got) {
+				t.Errorf("want len = %v, got = %v", tc.wantLen, len(got))
+			}
```

Comment:
This test doesn't feel like it's doing a particularly thorough job - it's checking the length but not the contents? I'm a little concerned it might be too brittle 

Would this be a better candidate for an e2e test?

Reply:
I agree -- this is a feeble test. I'll add an e2e test in the next PR and remove this.